File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int,int>mp;
        for(int i=0;i<n;i++){
            int a = target - nums[i];
            if(mp.find(a) != mp.end()){
                return {i,mp[a]};
            }
            mp[nums[i]]=i;
        }
      return {-1,-1};
        
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
       n = len(nums)
       mp = {}
       for i, num in enumerate(nums):
           a = target - num
           if a in mp:
              return [i,mp[a]]
           mp[num] = i; 
       return [-1,-1]
        